redis/commands/search/query.py

```diff
@@ -194,7 +194,7 @@ def _get_args_tags(self):
             args += self._ids
         if self._slop >= 0:
             args += ["SLOP", self._slop]
-        if self._timeout:
+        if self._timeout is not None:
```

Comment:
If we need positive timeouts... let's enforce @meiravgri

Reply:
Done and also updates the test

Reply:
We could have used string until this change, no? Do we want to handle strings here?

Reply:
Can’t we just rely on the server to validate the input (like it was up until today)?
becuase today we support both non negative int & **strings** of non negative ints
It makes the validation in the client more **complicated** and **slow**
Also, we don’t need to **make sure we are aligned** with redis on the definition of valid input (lets say we will want to change unlimited timeout to -1 instead of 0 (in redis), it can take time and pain until we will notice the client needs to be fixed too)

Reply:
I agree. Also IMHO in the cases where these errors are well-handled in the server, we should assume that most requests won't have syntax errors and the flow should be optimized for them. In the worst case, the syntax error will be caught anyway on the server side and will raise an error containing the problem.

Reply:
And regardless, I think that this PR was meant to fix a bug. Let's make more significant changes separately.

Reply:
We have varying states in the library where we do this - and in general it would be great if a low-level library provided this support. I'm of the opinion that if there's a requirement for a positive integer, let's give the user a useful exception, as [we do for ACL](https://github.com/redis/redis-py/blob/1596ac6a0bbc1e00e57c6a48e255a8f917498e39/redis/commands/core.py#L347C10-L347C10).

The clients need to be better than this.

Reply:
@dvora-h What do you think ? :)

Reply:
I'm with @meiravgri and @GuyAv46.
@chayim In most cases we do not validate the input, it's a different discussion if we want to change it or not and out of the scope here.

And yes, @meiravgri it's important to support also string inputs so you need to change/delete (IMO delete) the validation check